services/AttendanceService.ts

```diff
@@ -63,22 +63,30 @@ export default class AttendanceService {
     return Repositories.attendance(txn).attendEvent(user, event, attendedAsStaff);
   }
 
-  public async submitAttendanceForUser(userUuid: Uuid, eventUuid: Uuid, asStaff = false,
-    admin: UserModel): Promise<PublicAttendance> {
+  public async submitAttendanceForUsers(emails: string[], eventUuid: Uuid, asStaff = false,
+    admin: UserModel): Promise<PublicAttendance[]> {
     return this.transactions.readWrite(async (txn) => {
       const event = await Repositories.event(txn).findByUuid(eventUuid);
       if (!event) throw new NotFoundError('This event doesn\'t exist');
 
-      const user = await Repositories.user(txn).findByUuid(userUuid);
-      if (!user) throw new NotFoundError('This user was not found');
+      const users = await Repositories.user(txn).findByEmails(emails);
```

Comment:
`Map<String, Maybe<UserModel>>` might be a more useful data type for `UserRepository::findByEmails` to return.

```typescript
const usersByEmails: Map<String, Maybe<UserModel>> = await Repositories
  .user(txn)
  .findByEmails(emails);
const emailsNotFound: string[] = usersByEmails.entries()
  .filter([_email, u] => u.isNone())
  .map([email, _u] => email);
if (emailsNotFound.length > 0) {
  throw new BadRequestError('Couldn\'t find accounts matching one or more emails', {
    emails: emailsNotFound,
  });
}
```

Reply:
I can't actually find an official `Maybe<T>` type though.

Reply:
There's an `Option<T>` in the [`fp-ts`](https://www.npmjs.com/package/fp-ts) package that we can use.

Reply:
I guess email is part of `UserModel` though, so we could just do
```typescript
const users: UserModel[] = await Repositories.user(txn).findByEmails(emails);
const foundEmails: string[] = users.map(u => u.email);
const emailsNotFound = emails.filter(e => !foundEmails.contains(e));
```
but I like the idea of returning a type that's keyed by the search params.

Reply:
Lmk what you think.

Reply:
I also like the idea of returning the keyed param type. I think another way we can do it within SQL is running a query like:

```sql
SELECT emails.id, "Users".* FROM "Users" RIGHT JOIN (VALUES ('email1@ucsd.edu'), ('email2@ucsd.edu'), ('email3@ucsd.edu')) AS emails(id) ON "Users".email = emails.id;
```
which will return all the emails present in the `emails` array mapped to a user if it exists, otherwise null. Thoughts on this?

Reply:
IMO we should keep the query simple and do the mapping in the code. The goal of using an `Option<T>` type is to avoid nulls and have a type specifically for "this might or might not exist" since nulls can be finicky to work with and cause weird errors, so we'd want to return `Option.none()` instead of `null`.